overlord/devicestate/handlers_install.go

```diff
@@ -284,7 +284,16 @@ func buildInstallObserver(model *asserts.Model, gadgetDir string, useEncryption
 }
 
 func (m *DeviceManager) doSetupUbuntuSave(t *state.Task, _ *tomb.Tomb) error {
-	return m.setupUbuntuSave()
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	deviceCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return fmt.Errorf("cannot get device context: %v", err)
+	}
+
+	return m.setupUbuntuSave(deviceCtx)
```

Comment:
Just pointing out that now the whole `setupUbuntuSave()` function runs while the lock is being held. If this is intentional, please ignore this comment :-)

Reply:
given that we are still in the StartUp phase we can't process changes yet so shouldn't make a big difference